src/x/net/http/errors.go

```diff
@@ -92,25 +92,30 @@ func WriteError(w http.ResponseWriter, err error, opts ...WriteErrorOption) {
 		fn(&o)
 	}
 
+	statusCode := getStatusCode(err)
+	if o.response == nil {
```

Comment:
No reasponse does not meed there was an error; I think you should be checking the `err` here instead.

Reply:
`response` is set when there is a custom response error (instead of JSON encoding it).

See use here:
https://github.com/m3db/m3/blob/2c170ebd9c4acf6f132213eb07bf7c87a1077b14/src/query/api/v1/handler/prom/common.go#L102-L103

Reply:
(I think this logic is correct)

Reply:
My bad, missed the fact that this is inside `WriteError` function and not a part of some general response handling.